flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -88,13 +347,280 @@ def add_wind_forcing(
+                    engine="pyogrio",
+                ).to_crs(4326)
+                # keep only aggregation area chosen
+                polygon_file = aggr_areas.loc[
+                    aggr_areas[aggr_dict.field_name]
+                    == green_infrastructure.attrs.aggregation_area_name,
+                    ["geometry"],
+                ].reset_index(drop=True)
+        else:
+            raise ValueError(
+                f"The selection type: {green_infrastructure.attrs.selection_type} is not valid"
+            )
+
+        gdf_green_infra = self._model.data_catalog.get_geodataframe(
+            polygon_file,
+            geom=self._model.region,
+            crs=self._model.crs,
+        )
+
+        # Make sure no multipolygons are there
+        gdf_green_infra = gdf_green_infra.explode()
+
+        # Volume is always already calculated and is converted to m3 for SFINCS
+        height = None
+        volume = green_infrastructure.attrs.volume.convert(UnitTypesVolume("m3"))
+
+        # HydroMT function: create storage volume
+        self._model.setup_storage_volume(
+            storage_locs=gdf_green_infra, volume=volume, height=height, merge=True
+        )
+
+    def _add_measure_pump(self, pump: Pump):
+        """Add pump to sfincs model.
+
+        Parameters
+        ----------
+        pump : PumpModel
+            pump information
+        """
+        polygon_file = Database().input_path.joinpath(pump.attrs.polygon_file)
+        # HydroMT function: get geodataframe from filename
+        gdf_pump = self._model.data_catalog.get_geodataframe(
+            polygon_file, geom=self._model.region, crs=self._model.crs
+        )
+
+        # HydroMT function: create floodwall
+        self._model.setup_drainage_structures(
+            structures=gdf_pump,
+            stype="pump",
+            discharge=pump.attrs.discharge.convert(UnitTypesDischarge("m3/s")),
+            merge=True,
         )
 
-    def add_wind_forcing_from_grid(self, ds: xr.DataArray):
+    ### PROJECTIONS HELPERS - not needed at the moment ###
+
+    ##### OLD CODE BELOW #####
+    # @Gundula: I have renamed some the functions and added comments to some that explain when they were called in the original hazard.py just to keep track.
+    # Im not super familiar with hydromt_sfincs and its functions, so I have not changed the logic of the functions, only the names and comments.
+    # It could be the case that some of the functions in here are not needed anymore, but we can always remove them later if needed.
+
+    # I have also added a '_' to the start of all functions I see as private.
+    # Public functions can be called from outside of the class, all private functions should not be called from outside of the class.
+    # (even though python does not care and will allow it anyways, we can still follow the convention just to make it clear)
+
+    ### SFINCS SETTERS ###
+    def _set_timing(self, event: EventModel):
```

Comment:
I think set_timing should be generic to all hazard models (in addition to add_forcing, add_projection and add_measures)